router/functionHandler.go

```diff
@@ -168,6 +161,11 @@ func (roundTripper RetryingRoundTripper) RoundTrip(req *http.Request) (resp *htt
 
 	executingTimeout := roundTripper.funcHandler.tsRoundTripperParams.timeout
 
+	// wrap the req.Body with another ReadCloser interface.
+	if req.Body != nil {
+		req.Body = &fakeCloseReadCloser{req.Body}
+	}
```

Comment:
Can we do `defer req.Body....RealClose()` here instead of calling it each place that returns?